discovery_engine_core/web-api/src/models.rs

```diff
@@ -75,12 +83,22 @@ pub(crate) struct InteractionRequestBody {
 
 /// Unique identifier for the user.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Display)]
-pub(crate) struct UserId(Uuid);
+pub(crate) struct UserId(String);
+
+impl UserId {
+    fn new(user_id_str: &str) -> Result<Self, Error> {
```

Comment:
:art: it's already clear from the type that this is a str
```suggestion
    fn new(user_id: &str) -> Result<Self, Error> {
```